other/cordic.py

```diff
@@ -0,0 +1,168 @@
+# CORDIC algorithm info: https://en.wikipedia.org/wiki/CORDIC
+
+from enum import Enum
+
+SHIFT = 16
+SHIFT_BASE = 65536
+
+# (1/scale_factor) which scale factor is Π(cos(arctan(2^-i)))
+SCALE_CONSTANT = 1.64676
+
+#  The elementary angles for the lookup table. They correspond to
+#  arctan(2^-i) and then shifted left by 8 bits in an integer format
+ELEM_ANGLE = [
+    2949120,
+    1740967,
+    919789,
+    466945,
+    234379,
+    117304,
+    58666,
+    29335,
+    14668,
+    7334,
+    3667,
+    1833,
+    917,
+    458,
+    229,
+    115,
+    57,
+    29,
+    14,
+    7,
+    4,
+    2,
+    1,
+]
```

Comment:
WIll this work?  Tuples take less memory than lists and they are a signal to the reader that the contents will not change at runtime.

```suggestion
ELEM_ANGLE = (
    2949120,
    1740967,
    919789,
    466945,
    234379,
    117304,
    58666,
    29335,
    14668,
    7334,
    3667,
    1833,
    917,
    458,
    229,
    115,
    57,
    29,
    14,
    7,
    4,
    2,
    1,
)
```